api-inference-community/docker_images/timm/app/pipelines/image_classification.py

```diff
@@ -4,16 +4,26 @@
 import torch
 from app.pipelines import Pipeline
 from PIL import Image
+from timm.data import resolve_data_config
+from timm.data.transforms_factory import create_transform
+from timm.models.hub import load_model_config_from_hf
 
 
 class ImageClassificationPipeline(Pipeline):
     def __init__(self, model_id: str):
-        if "resnet50d" in model_id:
-            self.model = timm.create_model("resnet50d", pretrained=True).eval()
-        elif "dpn92" in model_id:
-            self.model = timm.create_model("dpn92", pretrained=True).eval()
-        else:
-            raise EnvironmentError("Cannot detect the correct arch for this model")
+
+        self.hf_cfg, self.arch = load_model_config_from_hf(model_id)
+        self.model = timm.create_model(f"hf_hub:{model_id}", pretrained=True)
+        self.transform = create_transform(
+            **resolve_data_config(self.hf_cfg, model=self.model)
```

Comment:
Updated to use `self.hf_cfg` here instead of `{}` alone, which I had previously. This way, if the user defines any parameters in their `config.json` that that override the default config for transforms (size, std, mean), they'll be used instead of the defaults.